server/sonar-server/src/test/java/org/sonar/server/usertoken/TokenGeneratorImplTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.usertoken;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TokenGeneratorImplTest {
+  TokenGeneratorImpl underTest = new TokenGeneratorImpl();
+
+  @Test
+  public void generate_different_tokens() {
+    // this test is not enough to ensure that generated strings are unique,
+    // but it still does a simple and stupid verification
+    String firstToken = underTest.generate();
+    String secondToken = underTest.generate();
+
+    assertThat(firstToken)
+      .isNotEqualTo(secondToken)
+      .hasSize(40);
+  }
+
+  @Test
+  public void hash_characteristics() {
+    String hash = underTest.hash(RandomStringUtils.randomAlphanumeric(100));
```

Comment:
Random test input is not recommended as it may break reproducibility. String should be inlined.